src/endpoints/authentication/jwt_auth.cpp

```diff
@@ -82,34 +83,32 @@ namespace ccf
     return tenant_id && tid && *tid == *tenant_id;
   }
 
-  struct VerifiersCache
+  struct PublicKeysCache
   {
-    static constexpr size_t DEFAULT_MAX_VERIFIERS = 10;
+    static constexpr size_t DEFAULT_MAX_KEYS = 10;
```

Comment:
Just a general question, is the `DEFAULT_MAX_KEYS` variable the keys retrieved from the `/.well-known/oauth/openid/jwks/` endpoints and they are just cached if the keys changed? Just trying to understand this bit of code and also curious on the max key size of 10, was it just an arbitrary max or is there more to it.

Reply:
It was (and is) an LRU cache limit, so it stores `N` most recent used keys on the verifier side, this is purely updated when used.

So if there's more keys, they will be re-created if all of them are being repetitively used. It actually may be worth increasing, `10` seems quite a small amount to me.